tests/Doctrine/Tests/Common/Proxy/ProxyGeneratorTest.php

```diff
@@ -396,6 +398,48 @@ public function testFinalClassThrowsException()
         $proxyGenerator->generateProxyClass($this->createClassMetadata(FinalClass::class, []));
     }
 
+    public function testPhp8UnionTypes()
+    {
+        if (PHP_VERSION_ID < 80000) {
+            $this->markTestSkipped('Requires PHP 8.0');
+        }
+
```

Comment:
I never used it myself, but apparently there is an annotation for this:

```suggestion
    /**
     * @requires PHP >= 8.0.0
     */
    public function testPhp8UnionTypes()
    {
```